pkg/controllers/customroute/custom_route_controller.go

```diff
@@ -0,0 +1,132 @@
+	corev1informers "k8s.io/client-go/informers/core/v1"
+	corev1listers "k8s.io/client-go/listers/core/v1"
+
+	configv1 "github.com/openshift/api/config/v1"
+	routev1 "github.com/openshift/api/route/v1"
+	configsetterv1 "github.com/openshift/client-go/config/clientset/versioned/typed/config/v1"
+	configinformers "github.com/openshift/client-go/config/informers/externalversions/config/v1"
+	configlistersv1 "github.com/openshift/client-go/config/listers/config/v1"
+	"github.com/openshift/cluster-authentication-operator/pkg/controllers/common"
+	"github.com/openshift/library-go/pkg/controller/factory"
+	"github.com/openshift/library-go/pkg/operator/events"
+
+	routeinformer "github.com/openshift/client-go/route/informers/externalversions/route/v1"
+	routev1lister "github.com/openshift/client-go/route/listers/route/v1"
+	"github.com/openshift/library-go/pkg/operator/resourcesynccontroller"
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type customRouteController struct {
+	destSecret     types.NamespacedName
+	componentRoute types.NamespacedName
+	ingressLister  configlistersv1.IngressLister
+	ingressSetter  configsetterv1.IngressInterface
+	routeLister    routev1lister.RouteLister
+	secretLister   corev1listers.SecretLister
+	resourceSyncer resourcesynccontroller.ResourceSyncer
+}
+
+func NewCustomRouteCrontroller(
+	componentRouteNamespace string,
+	componentRouteName string,
+	destSecretName string,
+	destSecretNamespace string,
+	ingressInformer configinformers.IngressInformer,
+	ingressSetter configsetterv1.IngressInterface,
+	routeInformer routeinformer.RouteInformer,
+	secretInformer corev1informers.SecretInformer,
+	eventRecorder events.Recorder,
+	resourceSyncer resourcesynccontroller.ResourceSyncer,
+	operatorClient v1helpers.OperatorClient,
+) factory.Controller {
+	controller := &customRouteController{
+		destSecret: types.NamespacedName{
+			Namespace: destSecretNamespace,
+			Name:      destSecretName,
+		},
+		componentRoute: types.NamespacedName{
+			Namespace: componentRouteNamespace,
+			Name:      componentRouteName,
+		},
+		routeLister:    routeInformer.Lister(),
+		ingressLister:  ingressInformer.Lister(),
+		ingressSetter:  ingressSetter,
+		secretLister:   secretInformer.Lister(),
+		resourceSyncer: resourceSyncer,
+	}
+	return factory.New().
+		WithInformers(
+			ingressInformer.Informer(),
+			secretInformer.Informer(),
+		).
+		WithSyncDegradedOnError(operatorClient).
+		WithSync(controller.sync).
+		ResyncEvery(1*time.Minute).
+		ToController("componentRouteSecretSyncController", eventRecorder)
+}
+
+func (c *customRouteController) sync(ctx context.Context, syncCtx factory.SyncContext) error {
+	ingressConfig, err := c.ingressLister.Get("cluster")
+	if err != nil {
+		return err
+	}
+
+	// Update ingressConfig
+	route, err := c.routeLister.Routes("openshift-authentication").Get("oauth-openshift")
+	if err != nil {
+		return err
+	}
```

Comment:
`route.Spec.Host` should most likely be set in this controller.

That would mean moving `metadataController.handleRoute` here, and the route object should be moved to bindata.

Reply:
Can do!